rules/manager_test.go

```diff
@@ -2357,6 +2357,39 @@ func TestUpdateWhenStopped(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestGroup_Eval_RaceCondition(t *testing.T) {
```

Comment:
[nit] I would clarify which race

```suggestion
func TestGroup_Eval_RaceConditionOnStoppingGroupEvaluationWhileRulesAreEvaluatedConcurrently(t *testing.T) {
```

Reply:
Done in https://github.com/grafana/mimir-prometheus/pull/714/commits/e829903745c2ee2f4a697045370f67946fc353a7